File: py/examples/openai_style.py
import os
from imbrace import ImbraceClient

def main():
    # 1. Khởi tạo Client giống OpenAI
    client = ImbraceClient(
        api_key="sk-imbrace-xxx", # Lấy từ IMBrace Dashboard
        env="develop"             # develop / sandbox / stable
    )

    try:
        print("--- IMBrace Python SDK Demo ---")

        # 2. Chat AI - Quản lý Folders
        folders = client.chat_ai.list_folders()
        print(f"User has {len(folders)} chat folders.")

        # 3. Chat AI - Tạo Prompt chuyên dụng cho Project này
        # Một feature cực mạnh vừa được em thêm vào SDK
        new_prompt = client.chat_ai.create_prompt({
            "command": "expert-dev",
            "name": "Expert Software Architect",
            "content": "You are a software architect with 20 years experience."
        })
        print(f"Created Prompt: {new_prompt.get('name')}")

        # 4. Chat Completion (Chuẩn nghiệp vụ AI)
        response = client.chat_ai.chat({
            "model": "gpt-3.5-turbo",
            "messages": [
                {"role": "system", "content": "You are a helpful assistant."},
                {"role": "user", "content": "Tell me about IMBrace features."}
            ]
        })
        print("AI Response:", response["choices"][0]["message"]["content"])

        # 5. Automation - Trigger Workflow
        # Gửi dữ liệu vào Workflow xử lý lead hoặc lưu database
        automation_res = client.workflows.trigger_flow("YOUR_FLOW_ID", {
            "source": "SDK_DEMO",
            "payload": response["choices"][0]["message"]["content"]
        })
        print("Automation Response:", automation_res)

    except Exception as e:
        print(f"An error occurred: {e}")

if __name__ == "__main__":
    main()

File: ts/examples/openai-style.ts
import { createImbraceClient } from "../src/index.js";

async function main() {
  // 1. Khởi tạo Client (Giống OpenAI)
  const client = createImbraceClient({
    apiKey: "YOUR_API_KEY", // Hoặc accessToken cho client-side
    env: "develop",         // 'develop', 'sandbox', hoặc 'stable'
  });

  try {
    console.log("--- IMBrace SDK Demo ---");

    // 2. Chat AI - Liệt kê Models
    const { data: models } = await client.chatAi.listModels();
    console.log(`Available Models: ${models.map(m => m.id).join(", ")}`);

    // 3. Sử dụng Custom Prompts (Vừa được thêm vào SDK)
    // Lấy prompt đã tạo sẵn trong IMBrace System
    const myPrompt = await client.chatAi.getPrompt("coding-assistant");
    console.log(`Using Prompt: ${myPrompt.name}`);

    // 4. Chat Completion (OpenAI-style)
    const response = await client.chatAi.chat({
      model: "gpt-4o",
      messages: [
        { role: "system", content: myPrompt.content },
        { role: "user", content: "Làm sao để tích hợp IMBrace SDK?" }
      ],
      stream: false
    });
    console.log("AI Response:", response.choices[0].message.content);

    // 5. Automation - Trigger Workflow
    // Giả sử bạn có một workflow xử lý lead từ Chat
    const flowResult = await client.workflows.triggerFlow("YOUR_FLOW_ID", {
      chat_id: "chat_123",
      content: response.choices[0].message.content
    });
    console.log("Workflow Triggered:", flowResult);

  } catch (error) {
    console.error("Error using SDK:", error);
  }
}

main();
